log4j-parent/pom.xml

```diff
@@ -154,6 +154,15 @@
     <httpcore.version>4.4.16</httpcore.version>
     <jna.version>5.16.0</jna.version>
 
+    <!-- Add 'requires static' for annotation library dependencies (defined as <scope>provided</scope> in this file)
+         to allow them to be visible at compile time to make the '-Xlint:classfile' check of javac pass.
+         Overrides parent's 'bnd-jpms-module-info' and hence re-defines '$[bnd-module-name];access=0' -->
+    <bnd-jpms-module-info>$[bnd-module-name];access=0;modules="
+      biz.aQute.bnd.annotation,
+      com.google.errorprone.annotations,
+      com.github.spotbugs.annotations,
+      org.osgi.annotation.bundle
```

Comment:
Can you add `org.osgi.annotation.versioning` too?

Reply:
Good point. I think this is automatically visible transitively through `org.osgi.annotation.bundle` but we should still add it explicitly as it is directly used.

Added it.